virttest/qemu_devices/qcontainer.py

```diff
@@ -2079,6 +2082,20 @@ def images_define_by_params(self, name, image_params, media=None,
         :param name: Name of the new disk
         :param params: Disk params (params.object_params(name))
         """
+        def _create_image_offline_snapshot():
+            """Create a offline snapshot for image file."""
+            base_img_filename = storage.get_image_filename(
+                    image_params, data_root)
+            img_format = image_params.get('image_format')
+            sn_name = 'vl.%s' % utils_misc.generate_random_string(6)
+            img_sn_filename = os.path.join('/var/tmp', sn_name)
```

Comment:
please use data_dir.get_tmp_dir() to get temp directory

Reply:
please move image process function to 'qemu_storage' module

Reply:
Have these suggestions too, thanks.

Reply:
Thanks, have updated.